apps/web/src/components/templates/VariableManager.tsx

```diff
@@ -10,18 +10,28 @@ interface VariableManagerProps {
   index: number;
   variablesArray: Record<string, any>;
   hideLabel?: boolean;
+  control?: any;
+  path?: string;
 }
 
 interface VariableComponentProps {
   index: number;
   template: string;
+  control?: any;
+  path?: string;
 }
 
-export const VariableComponent = ({ index, template }: VariableComponentProps) => {
-  const { control, watch } = useFormContext();
+export const VariableComponent = ({ index, template, control, path }: VariableComponentProps) => {
+  const formPrefix = path ? `${path}variables.${index}` : `${template}.variables.${index}`;
```

Comment:
Just to double check that the intention is to create the `formPrefix` for the case where `path` exists as: ```${path}variables.${index}``` and not ```${path}.variables.${index}```.

Reply:
Yes. Because I want to allow thw option of path being ''. 
I want to refactor all the hooks and components related to var manager, we can do it much better. But will probably do it in cool down